### PR TITLE
Dimension of an ideal

### PR DESCRIPTION
**Pull request by ederc**
Allows AlgebraicSolving to compute the dimension of an ideal, if applicable.

### CODE REVIEW

File: src/Rings/mpoly-ideals.jl [hunk 1]
```diff
@@ -1885,7 +1885,11 @@ julia> dim(I)
     return I.dim
   end
   is_zero(ngens(base_ring(I))) && return 0 # Catch a boundary case
-  I.dim = Singular.dimension(singular_groebner_generators(I, false, true))
+  if (is_f4_applicable(I, degrevlex(base_ring(I))))
```
Comment (lgoettgens) [inline src/Rings/mpoly-ideals.jl]:
```suggestion
  if is_f4_applicable(I, degrevlex(base_ring(I)))
```

File: src/Rings/mpoly-ideals.jl [hunk 2]
```diff
@@ -1884,8 +1884,13 @@ julia> dim(I)
   if I.dim > -1
     return I.dim
   end
+  is_one(I) && return -1 # Catch a boundary case
```
Comment (lgoettgens) [inline src/Rings/mpoly-ideals.jl:1887]: Maybe stupid question: if I remember correctly, this is_one call already computes a groebner basis (possibly via singular). Wouldn't it then be more efficient to use this knowledge in Singular.dimension? Or does AlgebraicSolving.dimension make use of that as well?

Comment (ederc) [reply src/Rings/mpoly-ideals.jl:1887]: You are right, `is_one()`  computes a GB via Singular. I will work on this tomorrow, there is more that can be optimized, e.g. caching GBs.

**Comment by ederc** [comment]
Let me put this in draft mode, we first need to have #4246 and some discussions on default options before it makes sense to optimize this.

**Comment by afkafkafk13** [comment]
@ederc In view of the changes to the file structure introduced in the meantime in  #4250 , how do you intend to proceed with this PR?

**Comment by fingolfin** [comment]
@ederc what is the status of this PR?

**Comment by ederc** [comment]
This depends on decisions to be made w.r.t. GB handling, I still have this on my list.

**Pull request closed without merging**